### PR TITLE
feat: Drop in Attachments

### PR DESCRIPTION
**Pull request by lesquive**
**Description:**
This PR enables file drag-and-drop functionality within the Chat window. When a user drags a file over the designated drop zone, the component's background color dynamically changes to offer visual feedback. This functionality leverages the capabilities of the [react-dropzone](https://www.npmjs.com/package/react-dropzone) library.

accepted files:
```
    accept: {
      "image/*": [".png", ".jpeg", ".jpg", ".gif"],
    },
```

Screen Recording 2023-11-08 at 20 14 01

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by berkus** [comment]
@lesquive hi, please resolve the conflicts!

**Comment by lesquive** [comment]
@berkus merge conflicts fixed

File: src/components/conversations/ConversationContainer.tsx [hunk 1]
```diff
@@ -69,6 +71,10 @@ const ConversationContainer: React.FC<ConvoContainerProps> = (
     return;
   }, [props.conversation?.sid]);
 
+  const handleDropedFiles = (droppedFiles: File[]) => {
```
Comment (berkus) [inline src/components/conversations/ConversationContainer.tsx:74]: drop**p**ed

File: src/components/message/MessageList.tsx [hunk 2]
```diff
@@ -127,6 +157,14 @@ const MessageList: React.FC<MessageListProps> = (props: MessageListProps) => {
     });
   }, [messages]);
 
+  useEffect(() => {
+    const abortController = new AbortController();
+    handleDropedFiles(files);
```
Comment (berkus) [inline src/components/message/MessageList.tsx:162]: dropped...

File: src/components/message/MessagesBox.tsx [hunk 3]
```diff
@@ -46,6 +46,7 @@ interface MessageProps {
   participants: ReduxParticipant[];
   lastReadIndex: number;
   use24hTimeFormat: boolean;
+  handleDropedFiles: (droppedFiles: File[]) => void;
```
Comment (berkus) [inline src/components/message/MessagesBox.tsx:49]: dropped

**Review by berkus — CHANGES_REQUESTED** [review]

**Review by berkus — APPROVED** [review]

**Pull request merged**